### PR TITLE
Add home page and URLs redirecting 

### LINKED ISSUE #62: Create Home Page
Create a home page (some basic info about our product) and set as the default page.

### PR DESCRIPTION
**Pull request by shayakrach**
Moving all job URLs to a new file to make code maintainable
and readable.
Adding home page and set as the default page since jobs page is for logged-in users.
Adding URL redirecting.

close #62, #67.

To make this PR ready  I have added 2 issues that follow it #71, #72 that will be fixed after demo 3.

### CODE REVIEW

**Review by edenmaor — APPROVED** [review]

**Review by assafg95 — APPROVED** [review]

**Comment by assafg95** [comment]
Looks great!

**Review by AshkenYariv — APPROVED** [review]
@shayakrach looks good, both the homepage and the tests!

**Comment by ormergi** [comment]
Hi Shay, please change the PR title to reflect what this PR does (most part is adding a home page)
and the change its description to reflect what changes its brings and why you need them.
Note about tests, they should be integral part of your PR, there is no need to mention it on the description or on commit body messages (other than the title).

**Review by ormergi — CHANGES_REQUESTED** [review]
First commit, 
The title is too general, this commit is about moving stuff to files so better title would be "Extract jobs URL's to file".
As for the commit body it should describe **why** you need this change for the most part, the what should be understood by the implementation.

File: templates/home.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{% extends 'base.html' %}
+
+{% block title %} Side Hustle{% endblock %}
+
+{% block content %}
+
+<h1>Welcome to SideHustle</h1>
+
+
+{% endblock %}
```
Comment (ormergi) [inline templates/home.html]: nit: missing newline here

**Review by ormergi — COMMENTED** [review]
second commit
Regarding the commit message body, there is no need to repeat the title (Add home page == creating home page)

**Review by ormergi — CHANGES_REQUESTED** [review]
3rd commit:
Correct me if I am wrong, but this commit is about changing the app url redirection behavior following the new home page that has been added, right?
If so this is not refactoring, w/o these changes the home page wont show and its a dead code,
thus this commit should be part of the commit that adds the home page.

**Comment by shayakrach** [comment]
> Hi Shay, please change the PR title to reflect what this PR does (most part is adding a home page) and the change its description to reflect what changes its brings and why you need them. Note about tests, they should be integral part of your PR, there is no need to mention it on the description or on commit body messages (other than the title).

Fixed @ormergi .

**Review by ormergi — CHANGES_REQUESTED** [review]
4th commit:
Consider moving the files renaming and moving tests around on a separate PR it is not related to adding a home pare right? :slightly_smiling_face: 

As for the home page tests, you can also move them to the commit that adds the new business logic, not mandatory for now..

**Comment by ormergi** [comment]
> Fixed @ormergi .

I think you better go though all my comments first :slightly_smiling_face:

**Comment by shayakrach** [comment]
> second commit Regarding the commit message body, there is no need to repeat the title (Add home page == creating home page)

Fixed @ormergi.

**Comment by shayakrach** [comment]
> > Fixed @ormergi .
> 
> I think you better go though all my comments first 🙂

Yes, I read all comments. 
Fixing now all the problems.

File: templates/home.html (hunk 1 above)
Comment (shayakrach) [reply templates/home.html]: The error message was deleted.
This is not relevant to the flake8 check.

**Comment by shayakrach** [comment]
> he title is too general, this commit is about moving stuff to files so better title would be "Extract jobs URL's to file".
> As for the commit body it should des

Fixed.

**Comment by shayakrach** [comment]
> Correct me if I am wrong, but this commit is about changing the app url redirection behavior following the new home page that has been added, right?
> If so this is not refactoring, w/o th

I changed to a better title which explains why it should be included in this PR.

**Comment by shayakrach** [comment]
> 4th commit: Consider moving the files renaming and moving tests around on a separate PR it is not related to adding a home pare right? 🙂
> 
> As for the home page tests, you can also move them to the commit that adds the new business logic, not mandatory for now..

But I have to test all the redirecting logic in the same PR, no?

File: jobs/views.py [hunk 2]
```diff
@@ -2,33 +2,91 @@
...
     template_name = 'jobs.html'
     ordering = ['-id']
 
+    def dispatch(self, *args, **kwargs):
+        # Check if user is authenticated
+        if not self.request.user.is_authenticated:
+            # Redirect them to the home page if not
+            return redirect('home')
+
+        # Render the template if they are
```
Comment (ormergi) [inline jobs/views.py]: The code explains its self **in this case**, remove those comments, all over.

**Comment by ormergi** [comment]
> But I have to test all the redirecting logic in the same PR, no?

Yes in general you need to test the logic this PR brings, here it adds home page and set to be the default.
If this was the PR description it would be much more easy to understand what this PR does.
Especially the 4th commit message, its confusing I didn't realized at first that all tests are new and necessary.

Next time keep all the related changes on the same commit, for example on this PR:
you could merge 3&4 commits, then you would have a commit that adds a new page and another one that changes the behavior of the app accordingly + tests for the new logic you added.

For the sake of progressing and merge this PR, I wont insist on merging commits for now
but for following PR's consider what we discussed here :slightly_smiling_face:

**Comment by ormergi** [comment]
Regarding the second commit message, its about setting the new home page as default, right?
If so, the commit title should express exactly that. And the commit body should explain *why* you need the change, and how its going to affect other components (like you did which is great).

Same goes for other commits on this PR, please revisit them and think how you can improve.

File: jobs/views.py (hunk 2 above)
Comment (shayakrach) [reply jobs/views.py]: Fixed.

**Comment by shayakrach** [comment]
> > But I have to test all the redirecting logic in the same PR, no?
> 
> Yes in general you need to test the logic this PR brings, here it adds home page and set to be the default. If this was the PR description it would be much more easy to understand what this PR does. Especially the 4th commit message, its confusing I didn't realized at first that all tests are new and necessary.
> 
> Next time keep all the related changes on the same commit, for example on this PR: you could merge 3&4 commits, then you would have a commit that adds a new page and another one that changes the behavior of the app accordingly + tests for the new logic you added.
> 
> For the sake of progressing and merge this PR, I wont insist on merging commits for now but for following PR's consider what we discussed here 🙂

Yes, I understand it now.
I will be more clear and accurate in the next PR.
So is that okay for now @ormergi?

**Review by assafg95 — APPROVED** [review]

**Comment by assafg95** [comment]
Everything looks good!

**Review by AshkenYariv — APPROVED** [review]
Looks good!

**Review by ormergi — CHANGES_REQUESTED** [review]
First commit message: typo "Add" -> added

Second commit message, this part is confusing "..instead of the jobs page since
this page should only be visible to logged-in users".
Which page is "this page"? please be explicit with the page you refer to (e.g: home page, job page..) and change it accordingly.

Last commit: Remove this line "Refactoring jobs test."

Maybe I wasn't clear about the action items (AI), change the PR description and title to reflect what this PR does (Adding a home pare and set it as default since jobs page is for logged it users, no refactoring..).

**Comment by shayakrach** [comment]
> First commit message: typo "Add" -> added
> 
> Second commit message, this part is confusing "..instead of the jobs page since this page should only be visible to logged-in users". Which page is "this page"? please be explicit with the page you refer to (e.g: home page, job page..) and change it accordingly.
> 
> Last commit: Remove this line "Refactoring jobs test."
> 
> Maybe I wasn't clear about the action items (AI), change the PR description and title to reflect what this PR does (Adding a home pare and set it as default since jobs page is for logged it users, no refactoring..).

Fixed.

**Comment by ormergi** [comment]
Thanks!

**Pull request merged**